### ISSUE TITLE
Nextcloud 27 compatibility

### ISSUE THREAD

**Issue by mueslo** [nextcloud/maps#1069]
Nextcloud 27.0.0 was released today. However, none of the `maps` releases currently available have NC27 listed as a compatible release. It would be good to get either a compatible release or a timeline for a compatible release.

**Comment by hugalafutro** [comment]
Not compatible with 26.0.3 either:
```
A new version is available: Nextcloud 26.0.3
3 apps have no compatible version for this Nextcloud version available.

Apps missing compatible version 
[Extract ↗](https://apps.nextcloud.com/apps/extract)
[Maps ↗](https://apps.nextcloud.com/apps/maps)
[Video converter ↗](https://apps.nextcloud.com/apps/video_converter)
```

**Comment by louim-lbs** [comment]
I just updated to 26.0.3. Maps wasn't shown as incompatible, and everything works fine, for now.

**Comment by hugalafutro** [comment]
@louim-lbs I edited my post, the incompability for Maps only showed from the Nextcloud UI, once the actual upgrade started it suddenly wasn't listed for reasons unknown.

**Comment by tacruc** [comment]
Can you test the v1.1.0-nightly, then we can release something for NC27 soon.

**Comment by pwepwe973** [comment]
up

**Comment by DuneJeeper** [comment]
Upgraded to 27.0 Maps unable to enable.

**Comment by jessebot** [comment]
When I try to enable the maps app (version 1.1.0) on Nextcloud 27 via the web UI, I get:
```
An error occurred during the request. Unable to proceed.
Autoload path not allowed: /var/www/html/custom_apps/maps/appinfo/application.php
```

I'm using the nextcloud docker image if that helps. Happy to give other details that might be helpful.

**Comment by tacruc** [comment]
Can you test if deleting the ../appinfo/applications.php helps? The file is empty I don't know why the AIO container cares about it.

**Comment by DuneJeeper** [comment]
Running NC 27.0 on unRAID. Deleted everything related to maps under /apps. Installed again from NC apps. Same php error. Used config editor and php was not empty. 

`<?php
/**
 * ownCloud - maps
 *
 * This file is licensed under the Affero General Public License version 3 or
 * later. See the COPYING file.
 *
 * @author Sander Brand <brantje@gmail.com>, Vinzenz Rosenkranz <vinzenz.rosenkranz@gmail.com>
 * @copyright Sander Brand 2014, Vinzenz Rosenkranz 2016, 2017
 */

namespace OCA\Maps\AppInfo;


use OCA\Maps\DB\FavoriteShareMapper;
use \OCP\AppFramework\App;
use \OCP\IServerContainer;
use OCA\Maps\Hooks\FileHooks;
use OCA\Maps\Service\PhotofilesService;
use OCA\Maps\Service\TracksService;
use Symfony\Component\EventDispatcher\EventDispatcherInterface;


class Application extends App {
    public function __construct (array $urlParams=array()) {
        parent::__construct('maps', $urlParams);

        $container = $this->getContainer();

        $this->getContainer()->registerService('FileHooks', function($c) {
            return new FileHooks(
                $c->query(IServerContainer::class)->getRootFolder(),
                \OC::$server->query(PhotofilesService::class),
                \OC::$server->query(TracksService::class),
                $c->query(IServerContainer::class)->getLogger(),
                $c->query('AppName'),
                $c->query(IServerContainer::class)->getLockingProvider()
            );
        });

        $this->getContainer()->query('FileHooks')->register();

        $this->registerFeaturePolicy();
    }

	private function registerFeaturePolicy() {
		/** @var EventDispatcherInterface $dispatcher */
		$dispatcher = $this->getContainer()->getServer()->getEventDispatcher();

		$dispatcher->addListener('OCP\Security\FeaturePolicy\AddFeaturePolicyEvent', function (\OCP\Security\FeaturePolicy\AddFeaturePolicyEvent $e) {
			$fp = new \OCP\AppFramework\Http\EmptyFeaturePolicy();
			$fp->addAllowedGeoLocationDomain('\'self\'');
			$e->addPolicy($fp);
		});
	}

}
`

**Comment by DuneJeeper** [comment]
Had 2 php's. application.php and app.php. Deleted both. Maps now an icon at top of NC. Doesn't load a map. "Your" of "your photos" is jumbled. Settings menu doesn't load.

**Comment by tacruc** [comment]
the app.php is important.

**Comment by DuneJeeper** [comment]
> the app.php is important.

Wouldn't load at all without both php's gone. Tried every combination. With both gone it appeared but was non-functional.  Running NC 27.0 docker on unRAID.

**Comment by alaskanpuffin** [comment]
@DuneJeeper I had to restart the docker container after deleting application.php and I was able to enable it.

**Comment by DuneJeeper** [comment]
> @DuneJeeper I had to restart the docker container after deleting application.php and I was able to enable it.

Followed your steps. Could not replicate. It let me enable. It never appeared so I could get to it. First went to disabled. I enabled from there, but nothing appears so I can launch it.

**Comment by tacruc** [comment]
I have problems to reproduce this Issue it would be helpful if someone finds the crutial difference between this instances where the issue occurs and where it doesn't:
On  this configuration I couldn't reproduce the Issue:
<details>

<summary> 
System 1
</summary>

```json
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "tempdirectory": "\/tmp\/",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "dbtype": "mysql",
        "version": "27.0.0.8",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "allow_local_remote_servers": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "loglevel": 3,
        "maintenance": false,
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\Krita"
        ],
        "appstore.experimental.enabled": true,
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 86400,
        "session_keepalive": true,
        "trashbin_retention_obligation": "auto",
        "knowledgebaseenabled": true,
        "logtimezone": "Europe\/Berlin",
        "default_phone_region": "AT",
        "cron_log": true,
        "log_rotate_size": 10857600,
        "theme": "",
        "updatechecker": false,
        "htaccess.RewriteBase": "\/",
        "check_for_working_wellknown_setup": true,
        "check_for_working_htaccess": true,
        "appstoreenabled": true,
        "updater.release.channel": "beta",
        "singleuser": false,
        "mysql.utf8mb4": true,
        "app_install_overwrite": [
            "phonetrack",
            "spreed",
            "calendar",
            "impersonate",
            "onlyoffice",
            "drawio",
            "qownnotesapi",
            "tasks",
            "news",
            "dicomviewer",
            "maps",
            "social",
            "cookbook",
            "checksum",
            "workflow_ocr",
            "metadata",
            "gpgmailer",
            "issuetemplate",
            "end_to_end_encryption",
            "camerarawpreviews",
            "twofactor_nextcloud_notification",
            "recognize",
            "facerecognition",
            "bookmarks",
            "workflow_script",
            "video_converter",
            "files_readmemd",
            "previewgenerator",
            "uppush",
            "files_photospheres",
            "fulltextsearch",
            "bookmarks_fulltextsearch",
            "fulltextsearch_elasticsearch",
            "files_fulltextsearch",
            "files_fulltextsearch_tesseract"
        ],
        "mail_sendmailmode": "smtp",
        "mail_smtpauthtype": "PLAIN",
        "memories.ffmpeg_path": "\/usr\/bin\/ffmpeg",
        "memories.ffprobe_path": "\/usr\/bin\/ffprobe",
        "memories.transcoder": "\/srv\/www\/nextcloud\/apps\/memories\/exiftool-bin\/go-vod-amd64",
        "memories.no_transcode": false,
        "memories.qsv": false,
        "memories.gis_type": 1,
        "mail_smtpsecure": "ssl",
        "mail_smtpauth": 1,
        "memories.exiftool_no_local": true,
        "memories.vod.path": "\/srv\/www\/nextcloud\/apps\/memories\/exiftool-bin\/go-vod-amd64",
        "memories.vod.ffmpeg": "\/usr\/bin\/ffmpeg",
        "memories.vod.ffprobe": "\/usr\/bin\/ffprobe"
    }
}
```
</details>

<details>
<summary> dev-docker </summary>

```json
{
    "system": {
        "debug": true,
        "profiler": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-extra",
                "url": "\/apps-extra",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-shared",
                "url": "\/apps-shared",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-writable",
                "url": "\/apps-writable",
                "writable": true
            }
        ],
        "allow_local_remote_servers": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1025",
        "skeletondirectory": "\/skeleton",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "nextcloud",
            "172.17.0.2",
            "localhost"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "sqlite3",
        "version": "27.0.0.8",
        "overwrite.cli.url": "http:\/\/localhost",
        "updater.release.channel": "git",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "lookup_server": "",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "loglevel": 2,
        "maintenance": false
    }
}
```

</details>

**Comment by jessebot** [comment]
After I deleted application.php as suggested, then the app showed an official update in the console (for some reason it was now showing as the old stable version), I clicked update and then enabled it and all is working now 🤷

<img width="995" alt="Screenshot from 2023-07-05 of the maps app in the nextcloud apps console showing enabled and at version 1.1.0" src="https://github.com/nextcloud/maps/assets/2389292/7090c85e-8d27-4d02-92e1-e91d1502062a">

I'm not entirely sure what caused this or how it was fixed as that file was indeed empty.

**Comment by Hans1Hansen** [comment]
Maybe a similar problem. 
Nextcloud 26.0.3. Cannot be updated to 27.0 because one app is not compatible, Maps 1.0.2
A coupe of hours ago I got the information that Maps 1.1.0 is available througt the internal Nextcloud notification. But when I go to apps, Maps cannot be updateted (showing no Update button)
Deactivating the app and reactivating does not change anything.
If I deactivate and delete the app, it cannot be installed again and is now completely missing in the internal appstore. 

Looks like my Nextcloud thinks that Maps is incompatible to my instance and does not show it as installable again. For I am using a VM i could reset to a snapshot with a working Maps, but with the older version, of course.
What can I do to force an update?

Cheers, Hans

**Comment by Hans1Hansen** [comment]
I ignored the app warning and upgraded NC to 27.0
The update worked, Maps looks like this:
![image](https://github.com/nextcloud/maps/assets/42319895/f660d505-c9d4-475b-b899-bdfe44079d15)
I can either delete the app or install an untested app
When I install and activate, Maps starts. But I am not sure, how stable this will work. Again, no available update for Maps is shown.
![image](https://github.com/nextcloud/maps/assets/42319895/e1522950-878b-4078-a913-1ee779b3fe88)

**Comment by tacruc** [comment]
@Hans1Hansen maps v1.1.0 requires at least PHP 8.1. Might this he the reason why no update is shown for you?

**Comment by Hans1Hansen** [comment]
@tacruc Bingo! That's right. I still use PHP version 8.0.28. NC 27 warns, but still works with it. Maps does not. Thank you very much! Your answer ended my senseless search for a solution.

**Comment by Hans1Hansen** [comment]
Info for people who try, too:
On NC 26 I updated to PHP 8.1. Nevertheless no update available for Maps 1.0.3
Updated to NC 27 either. Now Maps shows version 1.1.0.

**Comment by monkeyhybrid** [comment]
I just enabled Maps 1.1.0 on my Nextcloud 27 instance, and it appears to be working perfectly with no other intervention required.

For the record, I'm using PHP 8.1.

**Comment by ppfeufer** [comment]
I got it as an update on NC 26.0.2 with PHP 8.2. No issues. Still waiting on https://github.com/nextcloud/twofactor_admin before I update to NC26.0.3 or even 27 though

**Comment by Holger79** [comment]
I don't know if it helps but here is my observation, I am using the latest Nextcloud Docker 27 image.
I could not enable the app and always got the error: "Autoload path not allowed:"
After deleting the applications.php I was able to activate the app. After that I removed the app again and downloaded it again and now it could be activated the app despite the fact that the applications.php ist still  present.
